agent/handlers/v4/tmdsstate.go

```diff
@@ -161,7 +161,7 @@ func (s *TMDSAgentState) getTaskMetadata(v3EndpointID string, includeTags bool)
 					Path: task.GetNetworkNamespace(),
 					NetworkInterfaces: []*tmdsv4.NetworkInterface{
 						{
-							DeviceName: "",
+							DeviceName: "ethx",
```

Comment:
Why this change?

Reply:
Thanks!

Reply:
Some context:
The `DeviceName` is required for network latency / network packet loss faults. We are exposing this information (and other task network config) via AgentState to facilitate the fault injection handler to start/stop/check network fault.

Why we need a non-empty value here which is fake: because in this PR, we introduce a new check ([link](https://github.com/aws/amazon-ecs-agent/blob/02cb57659a882844033d0c18c7e666609347fb61/ecs-agent/tmds/handlers/fault/v1/handlers/handlers.go#L688-L691)) to see if the deviceName is empty. So if that's empty, these [unit test](https://github.com/aws/amazon-ecs-agent/blob/78a2bf0c7d3ebd3a13de3ac733af46dfb3816b18/agent/handlers/task_server_setup_test.go#L3695-L3708) will fail.